tests/frontend/test_circuitspecs_chip2.py

```diff
@@ -0,0 +1,373 @@
+            ops.BSgate(np.pi / 4, np.pi / 2) | (q[1], q[2])
+            ops.Rgate(np.pi) | (q[1])
+            ops.BSgate(np.pi / 4, np.pi / 2) | (q[1], q[2])
+            ops.Rgate(np.pi) | (q[0])
+            ops.Rgate(0) | (q[1])
+            ops.Rgate(np.pi) | (q[2])
+            ops.Rgate(-np.pi) | (q[3])
+
+            # corresponds to an identity on modes [4, 5, 6, 7]
+            ops.Rgate(0) | (q[4])
+            ops.BSgate(np.pi / 4, np.pi / 2) | (q[4], q[5])
+            ops.Rgate(0) | (q[4])
+            ops.BSgate(np.pi / 4, np.pi / 2) | (q[4], q[5])
+            ops.Rgate(0) | (q[6])
+            ops.BSgate(np.pi / 4, np.pi / 2) | (q[6], q[7])
+            ops.Rgate(0) | (q[6])
+            ops.BSgate(np.pi / 4, np.pi / 2) | (q[6], q[7])
+            ops.Rgate(np.pi) | (q[5])
+            ops.BSgate(np.pi / 4, np.pi / 2) | (q[5], q[6])
+            ops.Rgate(np.pi) | (q[5])
+            ops.BSgate(np.pi / 4, np.pi / 2) | (q[5], q[6])
+            ops.Rgate(0) | (q[4])
+            ops.BSgate(np.pi / 4, np.pi / 2) | (q[4], q[5])
+            ops.Rgate(0) | (q[4])
+            ops.BSgate(np.pi / 4, np.pi / 2) | (q[4], q[5])
+            ops.Rgate(0) | (q[6])
+            ops.BSgate(np.pi / 4, np.pi / 2) | (q[6], q[7])
+            ops.Rgate(0) | (q[6])
+            ops.BSgate(np.pi / 4, np.pi / 2) | (q[6], q[7])
+            ops.Rgate(0) | (q[5])
+            ops.BSgate(np.pi / 4, np.pi / 2) | (q[5], q[6])
+            ops.Rgate(np.pi) | (q[5])
+            ops.BSgate(np.pi / 4, np.pi / 2) | (q[5], q[6])
+            ops.Rgate(np.pi) | (q[4])
+            ops.Rgate(0) | (q[5])
+            ops.Rgate(np.pi) | (q[6])
+            ops.Rgate(-np.pi) | (q[7])
+
+            ops.MeasureFock() | q
+
+        assert program_equivalence(res, expected, atol=tol)
+
+    def test_interferometers(self, tol):
+        """Test interferometers correctly decompose to MZ gates"""
+        prog = sf.Program(8)
+        U = random_interferometer(4)
+
+        with prog.context as q:
+            ops.S2gate(0.5, 0) | (q[0], q[4])
+            ops.S2gate(0.5, 0) | (q[1], q[5])
+            ops.S2gate(0.5, 0) | (q[2], q[6])
+            ops.S2gate(0.5, 0) | (q[3], q[7])
+            ops.Interferometer(U) | (q[0], q[1], q[2], q[3])
+            ops.Interferometer(U) | (q[4], q[5], q[6], q[7])
+            ops.MeasureFock() | q
+
+        res = prog.compile("chip2")
+
+        expected = sf.Program(8)
+
+        with expected.context as q:
+            ops.S2gate(0.5, 0) | (q[0], q[4])
+            ops.S2gate(0.5, 0) | (q[1], q[5])
+            ops.S2gate(0.5, 0) | (q[2], q[6])
+            ops.S2gate(0.5, 0) | (q[3], q[7])
+            ops.Interferometer(U, mesh="rectangular_symmetric", drop_identity=False) | (q[0], q[1], q[2], q[3])
+            ops.Interferometer(U, mesh="rectangular_symmetric", drop_identity=False) | (q[4], q[5], q[6], q[7])
+            ops.MeasureFock() | q
+
+        expected = expected.compile(DummyCircuit())
```

Comment:
not clear where the MZ gates mentioned in the docstring come in

Reply:
`mesh="rectangular_symmetric"` causes the interferometer to decompose using `MZgate`'s. Will update the docstring.